### PR TITLE
fix for compatible with cita

### PR DESCRIPTION
**Pull request by jerry-yu**
fix for compatible with cita

### CODE REVIEW

File: src/evm/extmock.rs [hunk 1]
```diff
@@ -96,6 +96,15 @@ impl ext::DataProvider for DataProviderMock {
...
+        match self.db.get(address) {
+            Some(account) => {
+                account.balance == U256::zero() && account.code.is_empty() && account.nonce == U256::zero()
+            }
+            None => true,
+        }
+    }
+
+    fn exist(&self, address: &Address) -> bool {
         self.db.get(address).is_none()
```
Comment (mohanson) [inline src/evm/extmock.rs]: `is_some()`?

File: src/executive.rs [hunk 2]
```diff
@@ -319,7 +323,7 @@ fn create<B: DB + 'static>(
         &address,
         balance,
         // The init nonce for a new contract is one, see above documents.
-        U256::one(),
+        U256::zero(),
```
Comment (ouwenkg) [inline src/executive.rs]: Since we change this value from `1 => 0`, please remove the comment in previous line.

File: src/executive.rs [hunk 3]
```diff
@@ -254,11 +259,10 @@ fn call<B: DB + 'static>(
...
     request: &InterpreterParams,
 ) -> Result<evm::InterpreterResult, err::Error> {
-    debug!("call request={:?}", request);
-    // Ensure balance
-    if !request.disable_transfer_value && state_provider.borrow_mut().balance(&request.sender)? < request.value {
+    // Here not need check twice,becauce prepay is subed ,but need think call_static
+    /*if !request.disable_transfer_value && state_provider.borrow_mut().balance(&request.sender)? < request.value {
         return Err(err::Error::NotEnoughBalance);
-    }
+    }*/
```
Comment (ouwenkg) [inline src/executive.rs]: Del those comments?

File: src/state/state.rs [hunk 4]
```diff
@@ -154,7 +152,8 @@ impl<B: DB> State<B> {
     /// Get the merkle proof for a given account.
     pub fn get_account_proof(&self, address: &Address) -> Result<Vec<Vec<u8>>, Error> {
         let trie = PatriciaTrie::from(Arc::clone(&self.db), Arc::new(hash::get_hasher()), &self.root.0)?;
-        let proof = trie.get_proof(common::hash::summary(&address[..]).as_slice())?;
+        //let trie = PatriciaTrie::from(Arc::clone(&self.db), Arc::new(hash::HasherNull::new()), &self.root.0)?;
+        let proof = trie.get_proof(&address[..])?;
```
Comment (ouwenkg) [inline src/state/state.rs]: remove this comment?

File: src/state/state.rs [hunk 5]
```diff
@@ -229,7 +226,6 @@ impl<B: DB> State<B> {
 
     /// Add balance by incr for an account.
     pub fn add_balance(&mut self, address: &Address, incr: U256) -> Result<(), Error> {
-        debug!("state.add_balance a={:?} incr={:?}", address, incr);
```
Comment (ouwenkg) [inline src/state/state.rs]: Why delete those debug info? I think it must be very useful while detecting problems in the future.

File: src/evm/extmock.rs (hunk 1 above)
Comment (jerry-yu) [reply src/evm/extmock.rs]: yes

**Pull request merged**